### PR TITLE
if we get a timeout on showing the qr code lets cancel all subsagas and bail

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers 
cc: @MarcoPolo @mmaxim @patrickxb @maxtaco

### CODE REVIEW

File: shared/actions/engine/helper.js [hunk 1]
```diff
@@ -152,6 +160,11 @@ class EngineRpcCall {
         }
 
         if (incoming.finished) {
+          // Used just by device add for now. this is to fix a bug and i'm not sure this should apply generally
+          if (incoming.finished.error && this._finishedErrorShouldThrow) {
```
Comment (maxtaco) [inline shared/actions/engine/helper.js]: I'm not sure why this didn't work previously.  Seems like `yield call([this, this._cleanup], subSagaTasks)` would still be called (Eeven if an error) and then control would resume in the saga.

**Comment by mmaxim** [comment]
So the idea here is that if an RPC call returns an error, the JS engine helper sets a field saying there was an error and throws an exception. Then when the Saga ends, it throws another exception if this bool is set. So if I wrote some code that set this bool, this thing would throw twice if I didn't catch the exception from call?

File: shared/actions/engine/helper.js (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/engine/helper.js]: before it'd just wait for subsagas to finish themselves and wouldn't interrupt them

**Comment by chrisnojima** [comment]
@mmaxim no: before if the call finished, no matter what it'd wait for a running subsaga to finish. Now if you pass this flag , on error, it'll cancel the subsagas

**Comment by MarcoPolo** [comment]
Should we really throw? it seems annoying to add another branching behavior for no reason. We can already return the error from the finished result. I think we should just cleanup the rpcs and return

**Comment by chrisnojima** [comment]
It seems like you'd want to know this happened and make a decision.

**Comment by chrisnojima** [comment]
Marco can you update this and merge. I'm out

**Comment by chrisnojima** [comment]
@MarcoPolo updated

**Pull request merged**